src/java/org/apache/cassandra/cql3/PasswordObfuscator.java

```diff
@@ -35,6 +40,18 @@ public String obfuscate(String sourceString)
         if (passwordTokenStartIndex < 0)
             return sourceString;
 
-        return sourceString.substring(0, passwordTokenStartIndex + PASSWORD_TOKEN.length()) + OBFUSCATION_TOKEN;
+        return sourceString.substring(0, passwordTokenStartIndex + PASSWORD_TOKEN.length()) + " " + OBFUSCATION_TOKEN;
+    }
+
+    public static String obfuscate(String query, RoleOptions opts)
+    {
+        if (opts == null || query == null || query.isEmpty())
+            return query;
+
+        Optional<String> pass = opts.getPassword();
+        if (!pass.isPresent() || pass.get().isEmpty())
+            return query;
+        else
+            return query.replaceAll(pass.get(), PasswordObfuscator.OBFUSCATION_TOKEN);
```

Comment:
Also, we will have to update NEWS.txt and the docs because now only the password will be obfuscated

Reply:
Good call! I guess you meant CHANGES.txt. Let me know otherwise.

Reply:
Also NEWS.txt because there will be a visible change for the users. The entries will obfuscate only the password and the DCL statements won't be that useless anymore

Reply:
Oh right! bc this was release on 4.0. Right.